client/sidebar/header/actions/CreateRoomList.js

```diff
@@ -53,10 +56,12 @@ function CreateRoomList() {
 	const setModal = useSetModal();
 
 	const canCreateChannel = useAtLeastOnePermission(CREATE_CHANNEL_PERMISSIONS);
+	const canCreateTeam = useAtLeastOnePermission(CREATE_TEAM_PERMISSIONS);
 	const canCreateDirectMessages = usePermission('create-d');
 	const canCreateDiscussion = useAtLeastOnePermission(CREATE_DISCUSSION_PERMISSIONS);
 
 	const createChannel = useReactModal(setModal, CreateChannel);
+	const createTeam = useReactModal(setModal, CreateTeam);
```

Comment:
Why not refactor `useReactModal` like here?

https://github.com/RocketChat/Rocket.Chat/pull/21080/files#diff-c15dcafcf94ea74e0e9e57364940d7c475da0dd26fb28a22fa482ec8374f890cR39-R55

There will be performance issue that justifies passing `setModal` as injected dependency.

Reply:
I was waiting for that merge situation...

Reply:
... well, I will include that in here!